src/subScreen/pumpSimulatorScreen.js

```diff
@@ -0,0 +1,216 @@
+            return false;
+        }
+    }
+    ErrorOnBolusButton() {
+        return match.accessible.Button(this.language.pumpSimulatorSettingsScreen.ErrorOnBolus);
+    }
+    ErrorOnSuspendLabel() {
+        return match.accessible.Label(this.language.pumpSimulatorSettingsScreen.ErrorOnSuspend);
+    }
+    async _isErrorOnSuspend() {
+        try {
+            await expect(this.ErrorOnTempBasalButton()).toHaveValue('1');
+            return true;
+        } catch (error) {
+            return false;
+        }
+    }
+    ErrorOnSuspendButton() {
+        return match.accessible.Button(this.language.pumpSimulatorSettingsScreen.ErrorOnSuspend);
+    }
+    ErrorOnResumeLabel() {
+        return match.accessible.Label(this.language.pumpSimulatorSettingsScreen.ErrorOnResume);
+    }
+    async _isErrorOnResume() {
+        try {
+            await expect(this.ErrorOnTempBasalButton()).toHaveValue('1');
+            return true;
+        } catch (error) {
+            return false;
+        }
+    }
+    ErrorOnResumeButton() {
+        return match.accessible.Button(this.language.pumpSimulatorSettingsScreen.ErrorOnResume);
+    }
+    DeletePumpLabel() {
+        return match.accessible.Label(this.language.pumpSimulatorSettingsScreen.DeletePump);
+    }
+    SuspendDeliveryButton() {
+        return match.accessible.Button(this.language.pumpSimulatorSettingsScreen.SuspendDelivery);
+    }
+    ResumeDeliveryButton() {
+        return match.accessible.Button(this.language.pumpSimulatorSettingsScreen.ResumeDelivery);
+    }
+    /**
+     * @param {object} settings
+     * @param {boolean} settings.errorOnBolus
+     * @param {boolean} settings.errorOnTempBasal
+     * @param {boolean} settings.errorOnSuspend
+     * @param {boolean} settings.errorOnResume
+     * @param {number} settings.reservoirRemaining
+     * @param {number} settings.batteryRemaining
+     */
+    async Apply(settings) {
+        if (settings.errorOnBolus) {
+            await this.ErrorOnBolusOn();
+        } else if (!settings.errorOnBolus) {
+            await this.ErrorOnBolusOff();
+        }
+        if (settings.errorOnTempBasal) {
+            await this.ErrorOnTempBasalOn();
+        } else if (!settings.errorOnTempBasal) {
+            await this.ErrorOnTempBasalOff();
+        }
+        if (settings.errorOnResume) {
+            await this.ErrorOnResumeOn();
+        } else if (!settings.errorOnResume) {
+            await this.ErrorOnResumeOff();
+        }
+        if (settings.errorOnSuspend) {
+            await this.ErrorOnSuspendOn();
+        } else if (!settings.errorOnSuspend) {
+            await this.ErrorOnSuspendOff();
+        }
+        if (settings.batteryRemaining) {
+            await this.ApplyBatteryRemaining(settings.batteryRemaining);
+        }
+        if (settings.reservoirRemaining) {
+            await this.ApplyReservoirRemaining(settings.reservoirRemaining);
+        }
+    }
+    async ErrorOnBolusOn() {
```

Comment:
[nit] Consider collapsing into one `SetErrorOnBolus` function with a parameter to indicate on (true), off (false), or nil (no change). Then the above `Apply` collapses down nicely.

Reply:
yeah good idea